src/linker/pipeline.py

```diff
@@ -176,15 +179,27 @@ def write_spark_module(self) -> None:
 use rule terminate_spark from spark_cluster with:
     input: rules.all.input.final_output"""
             if self.config.computing_environment == "slurm":
-                for rule in ["start_spark_master", "start_spark_worker"]:
-                    module += f"""
-use rule {rule} from spark_cluster with:
+                module += f"""
+use rule start_spark_master from spark_cluster with:
+    resources:
+        slurm_account={self.config.slurm_resources['slurm_account']},
+        slurm_partition={self.config.slurm_resources['slurm_partition']},
+        mem_mb={self.config.spark_resources['slurm_mem_mb']},
+        runtime={self.config.spark_resources['runtime']},
+        cpus_per_task={self.config.spark_resources['cpus_per_task']},
+        slurm_extra="--output '{self.config.results_dir}/spark_logs/start_spark_master-slurm-%j.log'"
+use rule start_spark_worker from spark_cluster with:
     resources:
         slurm_account={self.config.slurm_resources['slurm_account']},
         slurm_partition={self.config.slurm_resources['slurm_partition']},
-        mem_mb={self.config.spark_resources['mem_mb']},
+        mem_mb={self.config.spark_resources['slurm_mem_mb']},
         runtime={self.config.spark_resources['runtime']},
         cpus_per_task={self.config.spark_resources['cpus_per_task']},
-        slurm_extra="--output '{self.config.results_dir}/spark_logs/{rule}-slurm-%j.log'"
+        slurm_extra="--output '{self.config.results_dir}/spark_logs/start_spark_worker-slurm-%j.log'"
+    params:
+        terminate_file_name=rules.terminate_spark.output,
+        user=os.environ["USER"],
+        cores={self.config.spark_resources['cpus_per_task']},
+        memory={self.config.spark_resources['mem_mb']}
```

Comment:
Ok, so you are using both `mem_mb` and `slurm_mem_mb` but I don't really get the practical difference (I mean, sure, one's buffered and one isn't, but why?)

Reply:
It's responsive to this comment--https://github.com/ihmeuw/linker/pull/84#discussion_r1561827157
We want to give the slurm job a little bit more memory than the worker itself.